depends: make fontconfig build under clang-16

fanquake opened the pull request:
Use the same workaround we've applied to qrencode, and other packages. Fontconfig not building is currently a blocker for fuzz/sanitizer infra upgrades (#27298).
    
For now, this is also more straightforward than bumping the package, which introduces more complexity/usage of gperf.

Closes: #27299.

hebasto commented:
Concept ACK.

hebasto commented:
> Looks like we can't drop the patch, as we'd still end up needing gperf, however just adapting our current patch to the new code also doesn't build, so we'll need to do more there. In any case, even if fontconfig builds, Qt doesn't work (out of the box) with clang-16, so this seems like a waste of time.

hebasto commented:
> Looks like we can't drop the patch, as we'd still end up needing gperf

Maybe include `gperf` in our build tools (like `bison`)?

> In any case, even if fontconfig builds, Qt doesn't work (out of the box) with clang-16, so this seems like a waste of time.

Fixed in #27314.

fanquake commented:
> I bumped from guix 1.3 to 1.4 but that didn't fix anything

How did you bump? Was this replacing the 1.3 bin with 1.4, or did you `guix pull`?

fanquake committed:
depends: make fontconfig build under clang-16

Use the same workaround we've applied to qrencode, and other packages.
Fontconfig not building is currently a blocker for fuzz/sanitizer infra
upgrades.

For now, this is also more straightforward than bumping the package,
which introduces more complexity/usage of gperf.

fanquake commented:
I've tested this with a depends build, with clang-16, and it seems to build everything.

hebasto commented on depends/packages/fontconfig.mk:
```diff
@@ -9,6 +9,7 @@ $(package)_patches=gperf_header_regen.patch
 define $(package)_set_vars
   $(package)_config_opts=--disable-docs --disable-static --disable-libxml2 --disable-iconv
   $(package)_config_opts += --disable-dependency-tracking --enable-option-checking
+  $(package)_cflags += -Wno-int-conversion -Wno-implicit-function-declaration
```
Is `-Wno-int-conversion` really required to compile with clang-16?

hebasto reviewed (commented):
Tested dce897f7ba3c8ae401a799a754cdea68a24a2048 on Ubuntu 23.04.

hebasto commented:
> depends: fontconfig 2.14.2

Update the title?

hebasto reviewed (approved):
ACK 9cbc1c279247800d79c5f6f95c0c2d8f387aac0a
